src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java

```diff
@@ -314,9 +314,14 @@ public static boolean isReceiverParameter(DetailAST parameterDefAst) {
      * @return the access modifier of the method/constructor.
      */
     public static AccessModifierOption getAccessModifierFromModifiersToken(DetailAST ast) {
-        final DetailAST modsToken = ast.findFirstToken(TokenTypes.MODIFIERS);
-        AccessModifierOption accessModifier =
-                getAccessModifierFromModifiersTokenDirectly(modsToken);
+        AccessModifierOption accessModifier;
+        if (ast.getType() == TokenTypes.ENUM_CONSTANT_DEF) {
+            accessModifier = AccessModifierOption.PUBLIC;
```

Comment:
Why we think it is always public?
Please share a bit more details, thanks in advance.

Reply:
Enum constants are always public, static, and final. This cannot be changed/overridden in any way.

Reply:
Yes, I read it as enum itself.